modules/ppcp-order-tracking/src/OrderTrackingModule.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * The order tracking module.
+ *
+ * @package WooCommerce\PayPalCommerce\OrderTracking
+ */
+
+declare(strict_types=1);
+
+namespace WooCommerce\PayPalCommerce\OrderTracking;
+
+use Dhii\Container\ServiceProvider;
+use Dhii\Modular\Module\ModuleInterface;
+use Exception;
+use Interop\Container\ServiceProviderInterface;
+use Psr\Container\ContainerInterface;
+use Psr\Log\LoggerInterface;
+use WC_Order;
+use WooCommerce\PayPalCommerce\OrderTracking\Assets\OrderEditPageAssets;
+use WooCommerce\PayPalCommerce\OrderTracking\Endpoint\OrderTrackingEndpoint;
+use WooCommerce\PayPalCommerce\WcGateway\Exception\NotFoundException;
+use WooCommerce\PayPalCommerce\WcGateway\Helper\PayUponInvoiceHelper;
+use WooCommerce\PayPalCommerce\WcGateway\Settings\Settings;
+use WooCommerce\PayPalCommerce\WcGateway\Settings\SettingsListener;
+
+/**
+ * Class OrderTrackingModule
+ */
+class OrderTrackingModule implements ModuleInterface {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function setup(): ServiceProviderInterface {
+		return new ServiceProvider(
+			require __DIR__ . '/../services.php',
+			require __DIR__ . '/../extensions.php'
+		);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @param ContainerInterface $c A services container instance.
+	 * @throws NotFoundException
+	 */
+	public function run( ContainerInterface $c ): void {
+		/**
+		 * The Settings.
+		 *
+		 * @var Settings $settings
+		 */
+		$settings = $c->get( 'wcgateway.settings' );
+
+		/**
+		 * The PUI helper.
+		 *
+		 * @var PayUponInvoiceHelper $pui_helper
+		 */
+		$pui_helper = $c->get( 'wcgateway.pay-upon-invoice-helper' );
+		if ( $pui_helper->is_pui_ready_in_admin() ) {
```

Comment:
If this module's `run` is always executed, calling this method (with `pui_product_status->pui_is_active()` inside) may be bad for performance, because it may execute API requests (when PUI is not available).

Reply:
Now with https://github.com/woocommerce/woocommerce-paypal-payments/commit/4e8379af022df302d4e7ed4f3245fdd5cb3ed2b6 it will first check the Country code, but I don't know what else could be done here to improve it cause on **every request** I need to know once PUI is activated for German country then automatically enable tracking. I can also check this on settings screen only but in that way you have to visit that screen to auto-enable the tracking setting. Not sure if that makes sense.

Reply:
One way could be to fire a hook [here](https://github.com/woocommerce/woocommerce-paypal-payments/blob/20880d8250542384bbdc9b3ebd8da7a39fc4d779/modules/ppcp-wc-gateway/src/Helper/PayUponInvoiceProductStatus.php#L99) to trigger the check. And also e.g. after onboarding, `woocommerce_paypal_payments_gateway_migrate` etc.